sdk/core/System.ClientModel/src/Convenience/KeyCredential.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+
+namespace System.ClientModel;
+
+public class KeyCredential
+{
+    private string _key;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KeyCredential"/> class.
+    /// </summary>
+    /// <param name="key">Key to use to authenticate with the Azure service.</param>
+    /// <exception cref="System.ArgumentNullException">
+    /// Thrown when the <paramref name="key"/> is null.
+    /// </exception>
+    /// <exception cref="System.ArgumentException">
+    /// Thrown when the <paramref name="key"/> is empty.
+    /// </exception>
+#pragma warning disable CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
+    public KeyCredential(string key) => Update(key);
+#pragma warning restore CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
+
+    // Note: this is GetValue instead of GetKey to allow consistent naming
+    // across credential types.  For example, for NamedKeyCredential, we would have
+    // GetValues with two out params to enable atomicity in reading credential values.
+    public string GetValue() => Volatile.Read(ref _key);
+
+    public void Update(string key)
+    {
+        if (key is null) throw new ArgumentNullException(nameof(key));
+        if (key.Length == 0) throw new ArgumentException("Value cannot be an empty string.", nameof(key));
```

Comment:
```suggestion
        if (String.IsNullOrEmpty(key)) throw new ArgumentException("Value cannot be an empty string.", nameof(key));
```

Reply:
Not sure that I agree that we should validate against whitespace.   Generally, null or empty is good enough.  

We can't validate the key content/format, in general.  I'd be good with considering whitespace valid, since we can't prove that it isn't valid for some service implementations.

Reply:
I think that I had copied in what we have today in AzureKeyCredential, but without calling through the Argument.cs class in an Azure namespace.  I can double check this.  Since AzureKeyCredential will inherit from this type, would you be concerned about changing the behavior of the Azure.Core type, @jsquire?

Reply:
Replacing with a call to ClientUtilities for now, but let me know if you want to further discuss behavior!

Reply:
> would you be concerned about changing the behavior of the Azure.Core type, @jsquire?

To clarify, my comment was in response to Chris' suggestion.  I'm in favor of keeping the behavior that we currently have, both here and in the Azure Core credentials.